pkg/runctl/controller.go

```diff
@@ -271,14 +271,8 @@ func (c *Controller) syncHandler(key string) error {
 		return nil
 	}
 	// fast exit
-	if pipelineRunAPIObj.GetDeletionTimestamp().IsZero() {
-		if pipelineRunAPIObj.Status.State == api.StateFinished {
-			return nil
-		}
-	} else {
-		if !utils.StringSliceContains(pipelineRunAPIObj.ObjectMeta.Finalizers, k8s.FinalizerName) {
-			return nil
-		}
+	if pipelineRunAPIObj.Status.State == api.StateFinished && !utils.StringSliceContains(pipelineRunAPIObj.ObjectMeta.Finalizers, k8s.FinalizerName) {
+		return nil
```

Comment:
why we remove it in the fast exit?

Reply:
If the pipeline run is deleted before it is touched the first time it has no finalizer set.
In this case we would set State to Finished but don't have Result set which is an inconsistent state.